docs/Containers/Pi-hole.md

```diff
@@ -1,127 +1,408 @@
 
-Default password is random, it can be changed by running:
 ```
-docker-compose exec pihole pihole -a -p myNewPassword
+~/IOTstack/services/pihole/pihole.env
 ```
 
-References:
+> There is nothing about old menu which *requires* the variables to be stored in the `pihole.env` file. You can migrate everything to `docker-compose.yml` if you wish.
 
-* [Pi-hole on GitHub](https://github.com/pi-hole/docker-pi-hole)
-* [Pi-hole on Dockerhub](https://hub.docker.com/r/pihole/pihole)
+Pi-hole's authoritative list of environment variables can be found [here](https://github.com/pi-hole/docker-pi-hole#environment-variables). Although many of Pi-hole's options can be set through its web GUI, there are two key advantages to using environment variables:
+
+1. If you ever need to reset Pi-hole by [erasing its persistent storage area](#cleanSlate), configuration options set using environment variables will persist while those set through the GUI may be lost; and
+2. On at least two occasions in its history, Pi-hole upgrades have had the effect of wiping configuration options set through the GUI, whereas options set using environment variables survived. 
+
+### <a name="adminPassword"></a>Admin password
+
+The first time Pi-hole is launched, it checks for the `WEBPASSWORD` environment variable. If found, the right hand side becomes the administrative password.
+
+You can set the value of `WEBPASSWORD` in the IOTstack menu by:
+
+1. Placing the cursor on top of "pihole".
+2. If Pi-hole is not already selected as a container, press <kbd>space</kbd> to select it.
+3. Press the right arrow, and then
+4. Choose "PiHole Password Options".
+
+From there, you have the choice of:
+
+* *Use default password for this build*
+
+	Choosing this option results in:
+
+	```yaml
+	- WEBPASSWORD=IOtSt4ckP1Hol3
+	```
+
+* *Randomise password for this build*
+
+	Choosing this option results in a randomly-generated password which you can find by inspecting your `docker-compose.yml`.
+
+* *Do nothing*
+
+	Choosing this option results in:
+
+	```yaml
+	- WEBPASSWORD=%randomAdminPassword%
+	```
+
+	which is a valid password string so "%randomAdminPassword%" will become the password.
+
+Regardless of which option you choose, you can always edit your `docker-compose.yml` to change the value of the environment variable. For example:
+
+```yaml
+- WEBPASSWORD=mybigsecret
+```
+
+It is important to realise that `WEBPASSWORD` only has any effect on the very **first** launch. Once Pi-hole has been run at least once, the value of `WEBPASSWORD` is ignored and any changes you make will have no effect.
+
+If `WEBPASSWORD` is **not** set on first launch, Pi-hole defaults to a randomly-generated password which you can discover after the first launch like this:
+
+```bash
```

Comment:
In regards to #547, could you change this and later code-blocks from `bash` to `console`?

Reply:
Done.